src/test/java/com/arpnetworking/tsdcore/sinks/KairosDbSinkTest.java

```diff
@@ -134,18 +135,46 @@ public void testPost() throws InterruptedException, IOException {
         Assert.assertEquals(expected, actual);
 
         // Verify that metrics has been recorded.
-        Mockito.verify(_mockMetricsFactory, Mockito.times(1)).create();
+        Mockito.verify(_mockMetricsFactory, Mockito.times(3)).create();
         Mockito.verify(_mockMetrics, Mockito.times(1)).incrementCounter("sinks/http_post/kairosdb_sink_test/success", 1);
         Mockito.verify(_mockMetrics, Mockito.times(1)).incrementCounter("sinks/http_post/kairosdb_sink_test/status/2xx", 1);
         Mockito.verify(_mockMetrics, Mockito.times(1)).setTimer(
                 Mockito.matches("sinks/http_post/kairosdb_sink_test/queue_time"),
                 Mockito.anyLong(),
-                Mockito.any());
-        Mockito.verify(_mockMetrics, Mockito.times(1)).startTimer("sinks/http_post/kairosdb_sink_test/request_latency");
-        Mockito.verify(_mockMetrics, Mockito.times(1)).stopTimer("sinks/http_post/kairosdb_sink_test/request_latency");
-        Mockito.verify(_mockMetrics, Mockito.times(1)).close();
+                Mockito.any(TimeUnit.class));
+        Mockito.verify(_mockMetrics, Mockito.times(1)).setTimer(
+                Mockito.matches("sinks/http_post/kairosdb_sink_test/request_latency"),
+                Mockito.anyLong(),
+                Mockito.any(TimeUnit.class));
+        Mockito.verify(_mockMetrics, Mockito.times(3)).close();
     }
 
+    @Test
+    public void testPostFailure() throws InterruptedException, IOException {
+         // Fake a successful post to KairosDb
+        _wireMock.register(WireMock.post(WireMock.urlEqualTo(PATH))
+                .willReturn(WireMock.aResponse()
+                        .withStatus(404)));
+        _kairosDbSinkBuilder.setMaximumRetries(1).build().recordAggregateData(com.arpnetworking.test.TestBeanFactory.createPeriodicData());
+
+        Thread.currentThread().join(5000);
```

Comment:
What is this for?

Reply:
^ I would instead suggest waiting on the mock server to have received the two requests. I don't know if WireMock has this built-in yet, but what I've done in the past is a register a handler with WireMock that decrements a semaphore or barrier and then have the main thread block on that with a timeout (the timeout would cause it to fail). You should probably also configure a smaller base backoff with this.